gunicorn/datadog_checks/gunicorn/gunicorn.py

```diff
@@ -53,21 +53,22 @@ def check(self, instance):
         proc_name = instance.get(self.PROC_NAME)
         master_procs = self._get_master_proc_by_name(proc_name, custom_tags)
 
-        # Fetch the worker procs and count their states.
-        worker_procs = self._get_workers_from_procs(master_procs)
-        working, idle = self._count_workers(worker_procs)
+        if master_procs:
+            # Fetch the worker procs and count their states.
+            worker_procs = self._get_workers_from_procs(master_procs)
+            working, idle = self._count_workers(worker_procs)
 
-        # if no workers are running, alert CRITICAL, otherwise OK
-        msg = "%s working and %s idle workers for %s" % (working, idle, proc_name)
-        status = AgentCheck.CRITICAL if working == 0 and idle == 0 else AgentCheck.OK
-        tags = ['app:' + proc_name] + custom_tags
+            # if no workers are running, alert CRITICAL, otherwise OK
+            msg = "%s working and %s idle workers for %s" % (working, idle, proc_name)
+            status = AgentCheck.CRITICAL if working == 0 and idle == 0 else AgentCheck.OK
+            tags = ['app:' + proc_name] + custom_tags
 
-        self.service_check(self.SVC_NAME, status, tags=tags, message=msg)
+            self.service_check(self.SVC_NAME, status, tags=tags, message=msg)
 
-        # Submit the data.
-        self.log.debug("instance %s procs - working:%s idle:%s", proc_name, working, idle)
-        self.gauge("gunicorn.workers", working, tags + self.WORKING_TAGS)
-        self.gauge("gunicorn.workers", idle, tags + self.IDLE_TAGS)
+            # Submit the data.
+            self.log.debug("instance %s procs - working:%s idle:%s", proc_name, working, idle)
+            self.gauge("gunicorn.workers", working, tags + self.WORKING_TAGS)
+            self.gauge("gunicorn.workers", idle, tags + self.IDLE_TAGS)
```

Comment:
```suggestion
            self.gauge("gunicorn.workers", idle, tags + self.IDLE_TAGS)
        else:
            self.log.debug("No master process(es), skipping worker metrics")
```

Should we log a helpful debug message when the check skips collecting `gunicorn.workers` metrics?

Reply:
definitely